snapshotter/demux/snapshotter.go

```diff
@@ -243,7 +243,8 @@ func (s *Snapshotter) Walk(ctx context.Context, fn snapshots.WalkFunc, filters .
 	contextLogger := log.G(ctx).WithField("function", "Walk")
 	namespace, err := getNamespaceFromContext(ctx, contextLogger)
 	if err != nil {
-		return err
+		contextLogger.Debug("no namespace found, proxying walk function to all cached snapshotters")
+		return s.cache.WalkAll(ctx, fn, filters...)
```

Comment:
What kind of errors can we expect from `Walk`? I think they would propagate from the provided `snapshots.WalkFunc`. Will these errors affect runtime or will they just be logged as transient?

Reply:
It seems like the errors are just being bubbled up to the parent caller: the return type of `WalkAll` in the interface signature is `error`.  I think this should be fine as-is.